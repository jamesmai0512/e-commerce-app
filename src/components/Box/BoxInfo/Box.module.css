.box_info {
  display: flex;
  flex-direction: column;
  gap: 15px;
  color: #2a254b;
  font-family: 'ClashDisplay Variable', sans-serif;
}

.box_info_description {
  font-size: 16px;
  line-height: 24px;
}

.box_info_title {
  padding: 16px 0 8px;
  font-size: 20px;
  line-height: 28px;
}
