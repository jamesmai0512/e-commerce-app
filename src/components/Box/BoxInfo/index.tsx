import { memo } from 'react'
import Image, { StaticImageData } from 'next/image'

// Styles
import styles from './Box.module.css'

export interface BoxInfoProps {
  icon?: StaticImageData | string
  title: string
  description: string
}
const BoxInfo = ({ icon = '', title, description }: BoxInfoProps) => {
  const { box_info_title, box_info_description } = styles
  return (
    <div className={styles.box_info}>
      <Image src={icon} alt="box-icon" width={24} height={24} loading="eager" />
      <h1 data-testid="box-title" className={box_info_title}>
        {title}
      </h1>
      <p className={box_info_description}>{description}</p>
    </div>
  )
}

export default memo(BoxInfo)
