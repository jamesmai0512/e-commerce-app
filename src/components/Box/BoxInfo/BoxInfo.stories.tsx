import React from 'react'
import { Meta, Story } from '@storybook/react'
import BoxInfo, { BoxInfoProps } from './'
import { ORDERSTEPS } from '../../../constants/common'

export default {
  title: 'Components/BoxInfo',
  component: BoxInfo,
} as Meta

const Template: Story<BoxInfoProps> = (args) => <BoxInfo {...args} />

export const WithIcon = Template.bind({})
WithIcon.args = {
  icon: ORDERSTEPS[1].icon,
  title: 'Title',
  description: 'Description',
}
