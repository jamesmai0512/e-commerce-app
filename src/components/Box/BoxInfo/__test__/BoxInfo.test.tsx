import { render } from '@testing-library/react'
import { StaticImageData } from 'next/image'
import BoxInfo from '../'

describe('BoxInfo component', () => {
  const mockIcon: StaticImageData = {
    src: 'https://raw.githubusercontent.com/jamesmai0512/image-ecommerce/main/i%1Dcons/ShoppingCart.png',
    height: 100,
    width: 100,
  }

  const BoxInfoProps = {
    icon: mockIcon,
    title: 'Test Title',
    description: 'Test description',
  }

  it('should render correctly', () => {
    const component = render(<BoxInfo {...BoxInfoProps} />)
    expect(component).toMatchSnapshot()
  })
})
