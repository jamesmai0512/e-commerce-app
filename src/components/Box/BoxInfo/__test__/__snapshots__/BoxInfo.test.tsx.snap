// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxInfo component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="box_info"
      >
        <img
          alt="box-icon"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="eager"
          src="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Fjamesmai0512%2Fimage-ecommerce%2Fmain%2Fi%251Dcons%2FShoppingCart.png&w=48&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Fjamesmai0512%2Fimage-ecommerce%2Fmain%2Fi%251Dcons%2FShoppingCart.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Fjamesmai0512%2Fimage-ecommerce%2Fmain%2Fi%251Dcons%2FShoppingCart.png&w=48&q=75 2x"
          style="color: transparent;"
          width="24"
        />
        <h1
          class="box_info_title"
          data-testid="box-title"
        >
          Test Title
        </h1>
        <p
          class="box_info_description"
        >
          Test description
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="box_info"
    >
      <img
        alt="box-icon"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="eager"
        src="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Fjamesmai0512%2Fimage-ecommerce%2Fmain%2Fi%251Dcons%2FShoppingCart.png&w=48&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Fjamesmai0512%2Fimage-ecommerce%2Fmain%2Fi%251Dcons%2FShoppingCart.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Fjamesmai0512%2Fimage-ecommerce%2Fmain%2Fi%251Dcons%2FShoppingCart.png&w=48&q=75 2x"
        style="color: transparent;"
        width="24"
      />
      <h1
        class="box_info_title"
        data-testid="box-title"
      >
        Test Title
      </h1>
      <p
        class="box_info_description"
      >
        Test description
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
